master/main.go

```diff
@@ -1,5 +1,113 @@
 package main
 
+import (
+	"flag"
+	"fmt"
+	"log"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/gorilla/websocket"
+	"gopkg.in/mgo.v2"
+	"gopkg.in/mgo.v2/bson"
+
+	"./models"
+)
+
+var connections map[*websocket.Conn]bool
+var initializeDB = false
+var dropDB = false
+var session *mgo.Session
+var collUsers *mgo.Collection
+
+func authentiaction(loginAndPass []string) bool {
+	result := models.User{}
+	err := collUsers.Find(bson.M{"username": loginAndPass[0]}).One(&result)
+	if err != nil {
+		return false
+	} else if result.Password == loginAndPass[1] {
+		return true
+	} else {
+		return false
+	}
+}
+
+func authHandler(w http.ResponseWriter, r *http.Request) {
+	if r.Header.Get("Origin") != "http://"+r.Host {
+		http.Error(w, "Origin not allowed", 403)
+		return
+	}
+
+	conn, err := websocket.Upgrade(w, r, w.Header(), 1024, 1024)
+	if err != nil {
+		http.Error(w, "Could not open websocket connection", http.StatusBadRequest)
+	}
+	for {
+		messageType, msg, err := conn.ReadMessage()
+		if err != nil {
+			return
+		}
+		data := strings.Split(string(msg), ",")
+		if authentiaction(data) {
+			conn.WriteMessage(messageType, []byte("true"))
+		} else {
+			conn.WriteMessage(messageType, []byte("false"))
+		}
+	}
+}
+
 func main() {
+	session, err := mgo.Dial("localhost")
+	if err != nil {
+		panic(err)
+	}
+	defer session.Close()
+
+	session.SetMode(mgo.Monotonic, true)
+
+	// drop database?
+	if dropDB {
+		err = session.DB("btrfs").DropDatabase()
+		if err != nil {
+			panic(err)
+		}
+	}
+
+	// Collection Users in BTRFS database
+	collUsers = session.DB("btrfs").C("users")
+
+	// database initialization? (Insert admin/admin user)
+	if initializeDB {
+		id := bson.NewObjectId()
```

Comment:
I'll just remove those dropDB and initializeDB - it can be done through mongo shell and in my opinion these operations don't need implementations in our code.